### PR TITLE
(dsl): Support `refresh` request

### LINKED ISSUE #262: Add support for refresh API
Add a way to explicitly refresh an index using the refresh API: https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-refresh.html

### PR DESCRIPTION
**Pull request by milicns**
This PR resolves #262

### CODE REVIEW

File: docs/overview/requests/elastic_request_refresh.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+id: elastic_request_refresh
+title: "Refresh Request"
+---
+
+This request refresh one or more Elasticsearch indices.
```
Comment (dbulaja98) [inline docs/overview/requests/elastic_request_refresh.md:6]: Maybe: This request is used for refreshing ...?

File: docs/overview/requests/elastic_request_refresh.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+You can create a `Refresh` request using the `refresh` method in the following manner:
+```scala
+// this import is required for using `IndexName`
+import zio.elasticsearch._
+
+val request: RefreshRequest = refresh(index = IndexName("index"))
+```
+
+
```
Comment (dbulaja98) [inline docs/overview/requests/elastic_request_refresh.md:22]: Omit one empty line.

File: modules/integration/src/test/scala/zio/elasticsearch/HttpExecutorSpec.scala [hunk 3]
```diff
@@ -634,6 +634,18 @@ object HttpExecutorSpec extends IntegrationSpec {
             }
           }
         ),
+        suite("refresh index")(
+          test("successfully refresh existing index") {
+            assertZIO(Executor.execute(ElasticRequest.refresh(index)))(
+              equalTo(true)
+            )
+          },
+          test("return 'NotFound' if index does not exists") {
```
Comment (dbulaja98) [inline modules/integration/src/test/scala/zio/elasticsearch/HttpExecutorSpec.scala:643]:
```suggestion
          test("return false if index does not exists") {
```

File: modules/library/src/main/scala/zio/elasticsearch/ElasticRequest.scala [hunk 4]
```diff
@@ -214,6 +214,17 @@ object ElasticRequest {
   final def getById(index: IndexName, id: DocumentId): GetByIdRequest =
     GetById(index = index, id = id, refresh = None, routing = None)
 
+  /**
+   * Constructs an instance of [[RefreshRequest]] used for deleting an index by specified name.
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/ElasticRequest.scala:218]: Update this scaladoc.

File: modules/library/src/test/scala/zio/elasticsearch/HttpElasticExecutorSpec.scala [hunk 5]
```diff
@@ -176,6 +176,16 @@ object HttpElasticExecutorSpec extends SttpBackendStubSpec {
 
         assertZIO(executorGetById)(isSome(equalTo(doc)))
       },
+      test("refresh") {
+        val executorRefresh =
+          Executor
+            .execute(
+              ElasticRequest
+                .refresh(name = index)
+            )
```
Comment (dbulaja98) [inline modules/library/src/test/scala/zio/elasticsearch/HttpElasticExecutorSpec.scala:185]: Can this be in one line?

**Review by dbulaja98 — CHANGES_REQUESTED** [review]

File: modules/library/src/main/scala/zio/elasticsearch/ElasticRequest.scala [hunk 6]
```diff
@@ -215,7 +215,7 @@ object ElasticRequest {
     GetById(index = index, id = id, refresh = None, routing = None)
 
   /**
-   * Constructs an instance of [[RefreshRequest]] used for deleting an index by specified name.
+   * Constructs an instance of [[RefreshRequest]] used for refreshing an index by specified name.
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/ElasticRequest.scala:218]:
```suggestion
   * Constructs an instance of [[RefreshRequest]] used for refreshing an index with specified name.
```

**Review by dbulaja98 — APPROVED** [review]

File: docs/overview/requests/elastic_request_refresh.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+You can create a `Refresh` request using the `refresh` method in the following manner:
+```scala
+// this import is required for using `IndexName`
+import zio.elasticsearch._
+
+val request: RefreshRequest = refresh(index = IndexName("index"))
+```
+
+You can find more information about `Refresh` request [here](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/indices-refresh.html).
```
Comment (dbulaja98) [inline docs/overview/requests/elastic_request_refresh.md:22]: Put empty line here.

**Review by dbulaja98 — CHANGES_REQUESTED** [review]

File: modules/library/src/main/scala/zio/elasticsearch/executor/HttpExecutor.scala [hunk 8]
```diff
@@ -648,6 +658,7 @@ private[elasticsearch] object HttpExecutor {
   private final val PointInTime   = "_pit"
   private final val Scroll        = "scroll"
   private final val ScrollId      = "scroll_id"
+  private final val Refresh       = "_refresh"
```
Comment (petarcurcin) [inline modules/library/src/main/scala/zio/elasticsearch/executor/HttpExecutor.scala:661]: Put this in right order.

**Review by petarcurcin — CHANGES_REQUESTED** [review]

**Review by petarcurcin — APPROVED** [review]

**Pull request merged**